### PR TITLE
cover ServiceImportResolver with unit test suite, fix 2 existing bugs

### PR DESCRIPTION
**Pull request by anadoba**
- added ServiceImportResolverSpec
- extracted test fixtures (custom services for testing) to a separate file that can be shared across different files
- found and fixed 2 bugs in ServiceImportResolver

### CODE REVIEW

File: postman-generator/src/test/scala/generator/TestFixtures.scala [hunk 1]
```diff
@@ -0,0 +1,147 @@
+package generator
+
+import io.apibuilder.spec.v0.models._
+
+object TestFixtures {
+
+  import models.TestHelper.{referenceApiService, generatorApiServiceWithUnionWithoutDescriminator}
+
+
```
Comment (ajanczak) [inline postman-generator/src/test/scala/generator/TestFixtures.scala]: mooooar space ;]

File: postman-generator/src/test/scala/generator/ServiceImportResolverSpec.scala [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    "leave the enums, models, unions type intact if it comes from the main service" in new ResolvedServiceTestContext {
+      val enumsIntact = mainService.enums.forall(enum => resultService.enums.exists(_.name == enum.name))
+      enumsIntact shouldEqual true
+
+      val modelsIntact = mainService.models.forall(model => resultService.models.exists(_.name == model.name))
+      modelsIntact shouldEqual true
+
+      val unionsIntact = mainService.unions.forall(union => resultService.unions.exists(_.name == union.name))
+      unionsIntact shouldEqual true
+    }
```
Comment (ajanczak) [inline postman-generator/src/test/scala/generator/ServiceImportResolverSpec.scala:49]:
```
 "leave the enums, models, unions type intact if it comes from the main service" in new ResolvedServiceTestContext {
      check(mainService.enums, resultService)
      check(mainService.models, resultService)
      check(mainService.unions, resultService)
    }

    def check[T](seq: Seq[T {def name: String}], resultService: Service) = {
      val unionsIntact = seq.forall(model => resultService.models.exists(_.name == model.name))
      unionsIntact shouldEqual true
    }
```

Comment (ajanczak) [reply postman-generator/src/test/scala/generator/ServiceImportResolverSpec.scala:49]: `check` name is just a placeholder ;)

File: postman-generator/src/test/scala/generator/ServiceImportResolverSpec.scala [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+      val modelNamesWithFullNamespace = importedService.models.forall { model =>
+        resultService.models.exists(m => m.name.contains(importedService.namespace) && m.name.contains(model.name))
+      }
+      modelNamesWithFullNamespace shouldEqual true
+
+      val unionNamesWithFullNamespace = importedService.unions.forall { union =>
+        resultService.unions.exists(u => u.name.contains(importedService.namespace) && u.name.contains(union.name))
+      }
+      unionNamesWithFullNamespace shouldEqual true
+    }
```
Comment (ajanczak) [inline postman-generator/src/test/scala/generator/ServiceImportResolverSpec.scala:67]: u can do the same here ^ ;)

**Comment by ajanczak** [comment]
Generally looks good 👍

**Review by ajanczak — APPROVED** [review]

**Comment by anadoba** [comment]
Applied all the comments

**Pull request merged**